src/components/navigation/components/LoginController.js

```diff
@@ -63,87 +63,103 @@ export class LoginController extends React.PureComponent {
     constructor(props){
         super(props);
         this.showLock = _.throttle(this.showLock.bind(this), 1000, { trailing: false });
-        this.loginCallback = this.loginCallback.bind(this);
-        this.loginErrorCallback = this.loginErrorCallback.bind(this);
+        this.validateCookieAndObtainAdditionalUserInfo = this.validateCookieAndObtainAdditionalUserInfo.bind(this);
+        this.auth0LoginCallback = this.auth0LoginCallback.bind(this);
         this.onRegistrationComplete = this.onRegistrationComplete.bind(this);
         this.onRegistrationCancel = this.onRegistrationCancel.bind(this);
         this.state = {
-            "isRegistrationModalVisible" : false,
-            "isLoading" : false // Whether are currently performing login/registration request.
+            // Contains email of Auth0-authenticated user but not in-system user
+            "unverifiedUserEmail" : false,
+            // Whether the code-split JS library for Auth0 has loaded yet.
+            // If false, is used to make Login/Register Button disabled temporarily.
+            "isAuth0LibraryLoaded": false,
+            // Whether are currently performing login/registration request.
+            "isLoading" : false,
+            // Held in here temporarily for user registration purposes.
+            // Should be deleted by time user can authenticate
+            "jwtToken": null
         };
     }
 
     componentDidMount () {
         const { auth0ClientID, auth0Domain, auth0Options } = this.props;
-
+        // prefetch & preload enabled here since is likely that user might want to click Login very quickly after loading webpage.
         import(
             /* webpackChunkName: "auth0-lock-bundle" */
             /* webpackMode: "lazy" */
+            /* webpackPrefetch: true */
+            /* webpackPreload: true */
```

Comment:
It's supposed to queue up the [code-split-out auth0 library JS file to be loaded in a tiny bit earlier than previously](https://webpack.js.org/guides/code-splitting/#prefetchingpreloading-modules). Not sure if it makes a practical difference since the auth0 library is queued to be loaded pretty early on anyway since is needed for the Login button to work.

Reply:
[Another relevant article](https://medium.com/webpack/link-rel-prefetch-preload-in-webpack-51a52358f84c)